messages/off-chain-registry/message_handler.go

```diff
@@ -84,17 +103,17 @@ func (m *messageManager) ShouldSendMessage(unsignedMessage *warp.UnsignedMessage
 		)
 		return false, err
 	}
-	if destination != m.registryAddress {
+	if destination != m.factory.registryAddress {
 		m.logger.Info(
 			"Message is not intended for the configured registry",
 			zap.String("destination", destination.String()),
-			zap.String("configuredRegistry", m.registryAddress.String()),
+			zap.String("configuredRegistry", m.factory.registryAddress.String()),
 		)
 		return false, nil
 	}
 
 	// Get the correct destination client from the global map
-	destinationClient, ok := m.destinationClients[destinationBlockchainID]
+	destinationClient, ok := m.factory.destinationClients[destinationBlockchainID]
```

Comment:
Would it make sense to pass the pointer to the destination client to the `ApplicationRelayer` instance, since each `ApplicationRelayer` only serves on src/dest pair? Then we don't have to deal with trying to grab it here.

Reply:
Good call out. I went ahead and added a destination client reference to the application relayer composition, and updated the message handler methods to accept the destination client directly.